mining.py

```diff
@@ -0,0 +1,175 @@
+import pickle
```

Comment:
Need file level comments on all your files, with your name and a description of your project